client.go

```diff
@@ -429,113 +428,95 @@ func (c *Client) Subscribe(intv time.Duration) (*Subscription, error) {
 }
 
 type PublishNotificationData struct {
-	Error                    error
-	SubscriptionID           uint32
-	DataChangeNotification   *ua.DataChangeNotification
-	EventNotificationList    *ua.EventNotificationList
-	StatusChangeNotification *ua.StatusChangeNotification
+	SubscriptionID uint32
+	Error          error
+	Value          interface{}
 }
 
-func (c *Client) Publish(dataChan chan<- PublishNotificationData) {
-	var (
-		subAck        = make([]*ua.SubscriptionAcknowledgement, 0)
-		pubRequest    = &ua.PublishRequest{}
-		pubResponse   *ua.PublishResponse
-		responseError ua.StatusCode
-		err           error
-	)
```

Comment:
We can reuse this vars in for loop for better performance.

Reply:
Did you test that this is actually the case or do you just assume that? Generally, you don't declare variables upfront in Go but declare them closer where they are being used. We also don't know whether the compiler is already optimizing this therefore you should profile this if you are concerned.